### ISSUE TITLE
SDK initialize errors when running tests in parallel

### ISSUE THREAD

**Issue by slevenick** [hashicorp/terraform-plugin-sdk#923]
### SDK version

```
2.10.1
```

### Relevant provider source code

Unrelated to provider source code. SDK error likely coming from here: https://github.com/hashicorp/terraform-plugin-sdk/blob/main/internal/plugintest/helper.go#L25

### Terraform Configuration Files

```hcl
...
```

### Debug Output

```
TF_ACC=1 TF_SCHEMA_PANIC_ON_ERROR=1 go test ./google-beta -parallel $ACCTEST_PARALLELISM -v -run=TestAcc -timeout 240m -ldflags="-X=github.com/hashicorp/terraform-provider-google-beta/version.ProviderVersion=acc"
... many tests above this line
--- SKIP: TestAccProviderIndirectUserProjectOverride (0.00s)
=== RUN   TestAccAccessApprovalSettings
=== RUN   TestAccAccessApprovalSettings/organization
=== RUN   TestAccAccessApprovalSettings/folder
cannot run Terraform provider tests: unexpected EOF
FAIL	github.com/hashicorp/terraform-provider-google-beta/google-beta	161.973s
FAIL
```

A similar error has been observed of `cannot run Terraform provider tests: Unknown status: 502` in other cases with the same effect.

These errors kill the process immediately

### Expected Behavior

Test runner should not crash causing the test process to immediately exit

### Actual Behavior

An error at the SDK level initializing a test seems to be killing the entire go test process

### Steps to Reproduce

Unclear exactly why these errors are occurring, as the errors are not very descriptive. Is there somewhere we should look for a better description of what is happening? These do not happen often, but may be linked to highly parallel (~32 parallel tests) execution of the tests? Seems to not be related to a specific test.

**Comment by detro** [comment]
Hello and thanks @slevenick for this issue report.

Looking at the `plugintest` package, the `error` is likely coming from the `DiscoverConfig` function. It's probably not super clear from the method doc, but internally that function will use environment variables and github.com/hashicorp/hc-install to attempt to install TF, in case the combination of environment variables suggests to do so.

This involves operations on the filesystem, as well as downloading (that `502` error above is telling).

In case of highly parallel tests, it can be hard to determine exactly what went wrong. But I'd like to try something: serialize and _cache_ the work of `DiscoverConfig`. Ultimately, it's operations are entirely based on the given source directory AND environment variables: so, when the source directory doesn't change, it shouldn't try to do this discovery more than once.

I'd also like to reproduce this error locally, but I'm new to the Google provider and I might need some pointers.

**Comment by detro** [comment]
Awesome!

**Comment by detro** [comment]
I'll close this now, but if the issue persists or comes back through a different path, please don't refrain to reach out. Here or in a new issue. 🙇‍♀️
